geode-core/src/main/java/org/apache/geode/cache/client/internal/pooling/AvailableConnectionManager.java

```diff
@@ -146,6 +146,11 @@ public boolean equals(Object o) {
       return false;
     }
```

Comment:
I don’t think there is even a reason this class implements equals. I’d say just let them be default.

Reply:
- This was implemented as a refactoring of the ConnectionManagerImpl as a part of improving performance.
- I believe that we cannot use the default implementation because it needs to store the first connection which was removed from the deque while executing  removeFirstOccurrence method in Deque.

Reply:
Oh yes, that is correct. I didn't notice this was that special inner class. Thanks!